### PR TITLE
Avoid creating rest elements with `errorType` when `any` is spread

### PR DESCRIPTION
**Pull request by Andarist**
This isn't particularly observable as far as I know. However, it turns out that currently `[...any]` is normalized~ to `[...errorType]`. `errorType` is a special `any` and displays as such, so this went unnoticed. However, since `[...any]` is legal it doesn't make sense to produce `errorType` in this situation at all.

EDIT:// this is more observable than what I've described above. I just pushed out a test for this. This PR fixes https://github.com/microsoft/TypeScript/issues/55932 that was incorrectly~ attributed to be a duplicate of https://github.com/microsoft/TypeScript/issues/29919

fixes https://github.com/microsoft/TypeScript/issues/57389

### CODE REVIEW

File: tests/baselines/reference/argumentsSpreadRestIterables(target=es5).types [hunk 1]
```diff
@@ -24,10 +24,10 @@ declare const itNum: Iterable<number>
 
 ;(function(a, ...rest) {})('', true, ...itNum)
 >(function(a, ...rest) {})('', true, ...itNum) : void
->(function(a, ...rest) {}) : (a: string, rest_0: boolean, ...rest_1: any[]) => void
->function(a, ...rest) {} : (a: string, rest_0: boolean, ...rest_1: any[]) => void
+>(function(a, ...rest) {}) : (a: string, rest_0: boolean, ...rest_1: Iterable<number>[]) => void
```
Comment (Andarist) [inline tests/baselines/reference/argumentsSpreadRestIterables(target=es5).types:27]: This test targets `es5` where `Iterable` doesn't exist. This change shows how now an unresolved type continues to be displayed with its alias name~. Just like here:
```ts
type Test = {
  foo: Thing; // even though it's unresolved and errors, it is still displayed as `Thing`
};
```

And actually, this matches the display of those IIFEs above this one. So it turns out that this one was going through `createNormalizedTupleType` that accidentally lost this display value.

**Comment by fatcerberus** [comment]
> normalized~

> incorrectly~

I still haven’t figured out how to interpret these tildes in your text

**Comment by Andarist** [comment]
😅 I use this as a softener of sorts - when I'm not sure if the used term is 100% precise etc ;p Sorry for making it confusing, I'm not sure where I got it from.

**Comment by sandersn** [comment]
Turns out this is a fix for a break introduced in https://github.com/microsoft/TypeScript/pull/57031 and reported a few days ago in #57389

@ahejlsberg @Andarist reported this as bug in #57389. Can you confirm and decide whether it's worth taking in 5.4?

**Comment by DanielRosenwasser** [comment]
@typescript-bot test top200

**Review by gabritto — APPROVED** [review]

**Comment by jakebailey** [comment]
@typescript-bot perf test this

**Review by DanielRosenwasser — APPROVED** [review]

**Comment by DavidArchibald** [comment]
This is also observable with something like this:
```ts
type SomeCallback<Args extends readonly any[]> = (...args: Args) => void;

function call<Callback extends SomeCallback<any>>(x: Callback, args: Parameters<Callback>) {
    x(...args);
}
```

Maybe this is just a simpler #55932 but I only knew how to fix this particular issue (which I actually ran into) because I've been reading the TypeScript issue tracker.

**Pull request merged**